firebase-vscode/webviews/components/ui/PanelSection.tsx

```diff
@@ -0,0 +1,42 @@
+import { VSCodeDivider } from "@vscode/webview-ui-toolkit/react";
+import React, { ReactNode, useState } from "react";
+import { Icon } from "./Icon";
+import { Spacer } from "./Spacer";
+import { Heading } from "./Text";
+import cn from "classnames";
+import styles from "./PanelSection.scss";
+
+export function PanelSection({
+  title,
+  children,
+  isLast,
+}: {
+  title?: ReactNode;
+  children: ReactNode;
+  isLast?: boolean;
+}) {
+  let [isExpanded, setExpanded] = useState(true);
+
+  return (
+    <>
+      {title && (
+        <button
+          aria-label={(isExpanded ? "Hide" : "Toggle") + " " + title}
+          className={cn(styles.panelExpando, isExpanded && styles.isExpanded)}
+          onClick={() => setExpanded(!isExpanded)}
+        >
+          <Icon className={styles.panelExpandoIcon} icon="chevron-down" />
+          <Heading level={5}>{title}</Heading>
+        </button>
+      )}
+      {isExpanded && (
```

Comment:
I don't think title needs to be part of the condition. It's expanded by default and if it doesn't have a title it doesn't have a title bar with a button that would let anyone ever collapse it.

Reply:
Without the extra logic, say if the title is initially non-empty and a user collapses a panel. Later if the panel title is changed to empty, the user will be left in a state where they can no longer see anything at all (since there's no title nor expand button).